src/main/java/org/spongepowered/common/world/SpongeExplosionBuilder.java

```diff
@@ -28,15 +28,29 @@
 import static com.google.common.base.Preconditions.checkState;
 
 import com.flowpowered.math.vector.Vector3d;
+import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
 import org.spongepowered.api.entity.explosive.Explosive;
 import org.spongepowered.api.world.Location;
 import org.spongepowered.api.world.World;
 import org.spongepowered.api.world.explosion.Explosion;
 import org.spongepowered.common.bridge.world.ExplosionBridge;
+import org.spongepowered.common.util.VecHelper;
+import org.spongepowered.api.world.explosion.ResistanceCalculator;
 
 public class SpongeExplosionBuilder implements Explosion.Builder {
 
+    private static final ResistanceCalculator DEFAULT_RESISTANCE_CALCULATOR =  ((blockState, blockPosition, explosion) ->
```

Comment:
More a remark: I'm not sure where this would go, or how I'd do this, but I feel like there should be a way to get the default calculator from the API...

Reply:
Good point, is there anyone you can think to ping for advice?

Reply:
Based on what I wrote in the API, I'd inline this again and then allow any calculator that exists to alter the value afterwards. Then the default doesn't need to be exposed and plugins can inspect what's going on before making a decision if necessary.

Reply:
If I inline it gets duplicated over two or three locations. That's why I created a static and just reference it inline.

Reply:
No, it wouldn't. 

> then allow any calculator that exists to alter the value afterwards

You'd ditch the default calculator, revert https://github.com/SpongePowered/SpongeCommon/pull/2640/files#diff-8b417570b299753ff70d99886c28d1a6L149-R159, allow the calculator to be nullable and do something like this after the initial calculation:

```
if (this.impl$resistance != null) {
    f2 = Math.min(0, Math.max(1, this.impl$resistance(f2, ...))); // (I get f2 is final at the moment, but this is the gist of what I was getting at)
}
```

Then a `null` resistance means default explosion resistance and it doesn't need to be exposed.

Reply:
```java
f2 = this.impl$resistanceCalculator.isPresent() ? this.impl$resistanceCalculator.get().calculateResistance( ... ) : <default behaviour>
```

Reply:
No. Can you please read what I said in the API issue (and above), reproduced below.

>    What about if a plugin wants to affect obsidian blocks only? How would they determine the resistance for other blocks?
>
>    Earlier, I said:
>
>  >    a ResistanceCalculator which takes a Location, the current resistance as a float, and returns the new resistance as a float.
>
>    That's the reason why I suggested it.

I'm not saying replace default behaviour. I'm saying do the default, then allow a plugin to modify it because a plugin might only want to affect a specific block and keep everything else as the default. Or they might want to multiply the resistance by some value.

Please follow what I've asked here, because I'm trying to keep as much flexibility for plugins as possible.

Reply:
So many individual parts over two threads got a little hard to keep track. Once I've figured out the solution I'll make a commit and we can try again.

It was intended that plugins could get the default if required but I can see now that I messed up somewhere while coding and I've got to spend a bit of time figuring out a solution.

Reply:
...please just do as I asked, calculate the default as it does now, and just add a step where it is passable to the calculator.

Reply:
But what if a plugin doesn't want to chain and instead ignore previous Calculators and use default resistance as it's base, maybe I've just gone down the rabbit hole and got lost?

Reply:
Just ignore the parameter?

Reply:
We did? I think I have a solution to it so that chaining can be done while still offering default values where required.